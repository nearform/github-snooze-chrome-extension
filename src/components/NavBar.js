import React, { useEffect, useState } from 'react'
import { useNavigate, useLocation } from 'react-router-dom'
import {
  AppBar,
  Box,
  Toolbar,
  Button,
  IconButton,
  Typography
} from '@mui/material'
import { Login, ChevronLeft, Settings } from '@mui/icons-material'
import { getCurrentRoute, routes } from '../routes'

function NavBar({ isAuthenticated }) {
  const navigate = useNavigate()
  const location = useLocation()

  const [showBackButton, setShowBackButton] = useState(false)
  const [currentRoute, setCurrentRoute] = useState(routes.dashboard)

  useEffect(() => {
    const locationUrl = location.pathname
    if (locationUrl !== routes.dashboard.url) {
      setShowBackButton(true)
      setCurrentRoute(getCurrentRoute(locationUrl))
    } else {
      setShowBackButton(false)
      setCurrentRoute(getCurrentRoute(locationUrl))
    }
  }, [location.pathname])

  const handleLogin = () => {
    navigate(routes.auth.url)
  }

  const handleNavigateBack = () => {
    navigate(-1)
  }

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar component="nav" position="fixed">
        <Toolbar>
          {showBackButton ? (
            <>
              <IconButton color="secondary" onClick={handleNavigateBack}>
                <ChevronLeft />
              </IconButton>
              <Typography variant="h6" component="span" sx={{ flexGrow: 1 }}>
                {currentRoute.title}
              </Typography>
            </>
          ) : (
            <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
              GitHub Snooze
            </Typography>
          )}
          {!showBackButton && (
            <Button
              color="secondary"
              variant="contained"
              endIcon={isAuthenticated ? <Settings/> : <Login />}
              onClick={handleLogin}
            >
              {isAuthenticated ? 'Settings' : 'Login' }
            </Button>
          )}
        </Toolbar>
      </AppBar>
      <Toolbar />
    </Box>
  )
}

export default NavBar
