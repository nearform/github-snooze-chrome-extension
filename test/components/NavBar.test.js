import React from 'react'
import renderer from 'react-test-renderer'
import NavBar from '../../src/components/NavBar'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useLocation: () => ({ pathname: '/' })
}))

describe('NavBar.js', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('shows the proper NavBar for unauthenticated users', async () => {
    let tree
    await renderer.act(async () => {
      tree = renderer.create(<NavBar isAuthenticated={false} />)
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('shows the proper NavBar for authenticated users', async () => {
    let tree
    await renderer.act(async () => {
      tree = renderer.create(<NavBar isAuthenticated={true} />)
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
