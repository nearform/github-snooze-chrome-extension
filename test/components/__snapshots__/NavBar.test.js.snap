// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.js shows the proper NavBar for authenticated users 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <nav
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        GitHub Snooze
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Settings
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </nav>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  />
</div>
`;

exports[`NavBar.js shows the proper NavBar for unauthenticated users 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <nav
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        GitHub Snooze
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Login
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LoginIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </nav>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  />
</div>
`;
